From f59d47398ce6261f96256121f171235e42f8c460 Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Sun, 28 Mar 2021 23:13:05 -0400
Subject: [PATCH] arm32: kaslr: Print the real kaslr offset when kernel panic
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f59d47398ce6261f96256121f171235e42f8c460


hulk inclusion
category: bugfix
bugzilla: 47952
CVE: NA

------------------------------------------------------------------------

The bss section is cleared when the kernel is started, and __kaslr_offset
variable is located in the bss section, __kaslr_offset is reset to zero,
so we move __kaslr_offset from bss section to data section.

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/kernel/head.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/kernel/head.S b/arch/arm/kernel/head.S
index b4666b8564bc..e5438d95ea30 100644
--- a/arch/arm/kernel/head.S
+++ b/arch/arm/kernel/head.S
@@ -103,11 +103,11 @@ ENTRY(stext)
 #ifdef CONFIG_RANDOMIZE_BASE
 	str_l	r3, __kaslr_offset, r9	@ offset in r3 if entered via kaslr ep
 
-	.section ".bss", "aw", %nobits
+	.pushsection .data		@ data in bss will be cleared
 	.align	2
 ENTRY(__kaslr_offset)
 	.long	0			@ will be wiped before entering C code
-	.previous
+	.popsection
 #endif
 
 #ifdef CONFIG_ARM_VIRT_EXT
-- 
2.26.2

